BaseError: column "coin_types" is of type numeric[] but expression is of type text[]

Blocks #53 

<details>
  <summary>Error log</summary>

```
4:40:55 AM ERROR indexing   Error while processing '/eth/base/Registry:NewResolver' event in 'base' block 26056917

BaseError: column "coin_types" is of type numeric[] but expression is of type text[]

    at handleNewResolver (/app/apps/ensnode/src/handlers/Registry.ts:227:26)

  225 |       } else {

  226 |         // otherwise upsert the resolver

> 227 |         const resolver = await upsertResolver(context, {

      |                          ^

  228 |           id: resolverId,

  229 |           domainId: node,

  230 |           address: resolverAddress,

Event arguments:

  node       0x47fa595ccffda0db5a5d2a63145ee3f24c855707b290f8327462e9151df3af9e

  resolver   0x0E37d13C0cb1ceCe1FB8c7b7ca912127198381A1

Log:

  index     274

  address   0xB94704422c2a1E396835A571837Aa5AE53285a95

Transaction:

  hash   0xb7c0768356e7a74c48d69cb4694ad0de9942a20f629d2fdd06c841596fa6dc20

  from   0xDd6d37E29294A985E49fF301Acc80877fC24997F

  to     0x0E37d13C0cb1ceCe1FB8c7b7ca912127198381A1

Block:

  hash        0xc5de7e0556df4fd2981447b67e87405c14a74d5d16341f87437f0717fe9e29df

  number      26056917

  timestamp   1738903181

4:40:55 AM WARN  process    Encountered indexing error, starting shutdown sequence

4:40:55 AM FATAL process    Finished shutdown sequence, terminating (exit code 1)

/app/apps/ensnode:

 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  ensnode@0.0.1 start: `ponder start "--schema" "d184809d-7be2-4982-96c7-805de13ea2e7"`

Exit status 1
```
</details>

--- a/apps/ensnode/src/handlers/Resolver.ts
+++ b/apps/ensnode/src/handlers/Resolver.ts
@@ -69,7 +69,7 @@ export const makeResolverHandlers = (ownedName: OwnedName) => {
       // upsert the new coinType
       await context.db
         .update(schema.resolver, { id })
-        .set({ coinTypes: uniq([...(resolver.coinTypes ?? []), coinType.toString()]) });
+        .set({ coinTypes: uniq([...(resolver.coinTypes ?? []), coinType]) });
 
       // log ResolverEvent
       await context.db.insert(schema.multicoinAddrChanged).values({

--- a/packages/ponder-schema/src/ponder.schema.ts
+++ b/packages/ponder-schema/src/ponder.schema.ts
@@ -106,11 +106,7 @@ export const resolver = onchainTable("resolvers", (t) => ({
   texts: t.jsonb().$type<string[]>(),
   // The set of observed SLIP-44 coin types for this resolver
   // NOTE: we avoid .notNull.default([]) to match subgraph behavior
-  // NOTE: we store coinTypes as a [String!], to avoid loss of precision due to drizzle parsing bug
-  // https://github.com/ponder-sh/ponder/issues/1475
-  // https://github.com/ponder-sh/ponder/pull/1482
-  // TODO: revert back to t.bigint().array()
-  coinTypes: t.text("coin_types").array(),
+  coinTypes: t.bigint().array(),
 }));
 
 export const resolverRelations = relations(resolver, ({ one, many }) => ({

